src/tests/ftest/server/multiengine_persocket.py

```diff
@@ -135,33 +134,112 @@ def daos_server_storage_prepare_ns(self, step, engines_per_socket=1):
         cmd.sudo = False
         cmd.debug.value = False
         cmd.set_sub_command("scm")
-        cmd.sub_command_class.set_sub_command("reset")
+        cmd.sub_command_class.set_sub_command("prepare")
         cmd.sub_command_class.sub_command_class.scm_ns_per_socket.value = engines_per_socket
         cmd.sub_command_class.sub_command_class.force.value = True
-        cmd.sub_command_class.sub_command_class.scm_only.value = True
 
         self.log.info(
-            "===(%s.B)Starting daos_server storage prepare -S: %s", step, str(cmd))
-        results = run_pcmd(self.hostlist_servers, str(cmd), timeout=180)
-        if results[0]['exit_status']:
+            "===(%s.B)Starting daos_server scm prepare -S: %s", step, str(cmd))
+        results = run_remote(self.log, self.hostlist_servers, str(cmd), timeout=180)
+        if not results.passed:
             self.fail(
                 "#({0}.B){1} failed, "
                 "please make sure the server equipped with {2} PMem "
                 "modules.".format(step, cmd, engines_per_socket))
-        time.sleep(15)
-        self.host_reboot(self.hostlist_servers)
+
+    def wait_for_result(self, get_method, timeout, **kwargs):
+        """Wait for a result with a timeout.
+
+        Args:
+            get_method (object): method to call to determine if the result is found
+            timeout (int): number of seconds to wait for a response to be found
+            kwargs (dict): kwargs for get_method.
+
+        Returns:
+            bool: if the result was found
+        """
+        result_found = False
+        timed_out = False
+        start = time.time()
+
+        while not result_found and not timed_out:
+            timed_out = (time.time() - start) >= timeout
+            result_found = get_method(**kwargs)
+            self.log.debug("==Response found by %s(%s): %s", get_method, kwargs, result_found)
+            time.sleep(5)
+        return not timed_out
+
+    def check_ping(self, host, expected_pings=0, cmd_timeout=60, verbose=True):
+        """Check the host for a ping response.
+
+        Args:
+            host (Node): destination host to ping to.
+            expected_pings (int, optional): number of succeed ping expected. default is 0.
+            cmd_timeout (int, optional): number of seconds to wait for a response to be found.
+            verbose (bool, optional): display check ping commands. Defaults to True.
+
+        Returns:
+            bool:  True if the expected number of pings were returned; False otherwise.
+        """
+        try:
+            result = run_local(
+                self.log, ["ping", "-c 1", str(host)], check=False,
```

Comment:
This is almost certainly wrong, I suspect you mean this:
```suggestion
                self.log, ["ping", "-c",  "1", str(host)], check=False,
```

How about passing the timeout onto the ping command with the -W option and using the returncode to determine success?

This function takes a "expected_pings" but only sends one ping so valid values that can be checked here are only 0 and 1 meaning the only thing this function can do if confirm that a host is up so you could remove complexity and use the return code.

Reply:
Please see the latest update with script fixes.